### PR TITLE
feat(web.domain): datagrid - behaviour rules

### PR DESCRIPTION
**Pull request by dectotam**
preparing future datagrid by settings columns and an api mockup ref: MANAGER-11839

| Question         | Answer
| ---------------- | ---
| Branch?          | feat/domain-optim
| Bug fix?         | no
| New feature?     | no
| Breaking change? | no
| Tickets          | Feat MANAGER-11839
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~~Breaking change is mentioned in relevant commits~~

## Description

Preparing future datagrid by settings columns and an api mockup

### CODE REVIEW

File: packages/manager/apps/web/client/app/domain/list/list-domain-layout.controller.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import punycode from 'punycode';
 import { ListLayoutHelper } from '@ovh-ux/manager-ng-layout-helpers';
+import get from 'lodash/get';
```
Comment (anooparveti) [inline packages/manager/apps/web/client/app/domain/list/list-domain-layout.controller.js:3]: **nitpick:** Avoid lodash and use native "Safe navigation operator" on JS

File: packages/manager/apps/web/client/app/domain/list/list-domain-layout.controller.js [hunk 2]
```diff
@@ -93,6 +78,82 @@ export default class ListDomainLayoutCtrl extends ListLayoutHelper.ListLayoutCtr
...
+          name: 'ns13.ovh.net',
+        },
+      ],
+    };
+    return this.$q.resolve({
+      data: [mockData].concat(get(this.resources, 'data')),
+      meta: {
+        totalCount: this.paginationTotalCount,
+      },
+    });
```
Comment (anooparveti) [inline packages/manager/apps/web/client/app/domain/list/list-domain-layout.controller.js:154]: **question:** Why mock data? Dont we have API?

File: packages/manager/apps/web/client/app/domain/list/template.html [hunk 3]
```diff
@@ -93,34 +93,19 @@
...
+    ></oui-datagrid-column>
+
     <oui-datagrid-column
-        data-title="::'domains_column_domains_whois_owner' | translate"
+        data-title="::'domains_column_domains_renewal_date' | translate"
+    ></oui-datagrid-column>
+
+    <oui-datagrid-column
+        data-title="::'domains_column_domains_expiration_date' | translate"
+    ></oui-datagrid-column>
```
Comment (anooparveti) [inline packages/manager/apps/web/client/app/domain/list/template.html:105]: Without `data-property`, the column will be empty

File: packages/manager/apps/web/client/app/domain/list/template.html [hunk 4]
```diff
@@ -137,6 +122,11 @@
             ></span
         ></a>
     </oui-datagrid-column>
+
+    <oui-datagrid-column
+        data-title="::'domains_column_domains_admin_contact' | translate"
+    ></oui-datagrid-column>
```
Comment (anooparveti) [inline packages/manager/apps/web/client/app/domain/list/template.html:128]: Without `data-property`, the column will be empty

File: packages/manager/apps/web/client/app/domain/list/template.html [hunk 5]
```diff
@@ -147,4 +137,8 @@
     >
         <span data-translate="{{ 'domains_dns_type_' + $value }}"></span>
     </oui-datagrid-column>
+
+    <oui-datagrid-column
+        data-title="::'domains_column_domains_name_server' | translate"
+    ></oui-datagrid-column>
```
Comment (anooparveti) [inline packages/manager/apps/web/client/app/domain/list/template.html:143]: Without `data-property`, the column will be empty

**Review by anooparveti — CHANGES_REQUESTED** [review]

File: packages/manager/apps/web/client/app/domain/list/template.html (hunk 3 above)
Comment (dectotam) [reply packages/manager/apps/web/client/app/domain/list/template.html:105]: It is normal : the columns completion are in other tickets, this is a just an initialization of the global structure of the datagrid.

File: packages/manager/apps/web/client/app/domain/list/list-domain-layout.controller.js (hunk 2 above)
Comment (dectotam) [reply packages/manager/apps/web/client/app/domain/list/list-domain-layout.controller.js:154]: Not yet

**Review by anooparveti — APPROVED** [review]

**Review by qpavy — APPROVED** [review]

**Pull request merged**